### PR TITLE
Add adaptiveCardsParserMaxVersion style options

### LINKED ISSUE #3777: Add a style options to select Adaptive Cards parser maximum supported version
## Feature Request

With introduction of Adaptive Cards 2.5.0 (support schema version 1.3), some developers are experiencing issues as they unexpectedly sending an Adaptive Cards with schema 1.3 previously.

Before Web Chat 4.12.0, we are using Adaptive Cards 1.2.x which support schema version 1.2. When a 1.3 card is parsed/rendered, it will render as a 1.2 card.

After Web Chat 4.12.0, we are using Adaptive Cards 2.5.0 which support schema version 1.3. A 1.3 card will be parsed/rendered as a 1.3 card.

If developers were unknowingly sending a 1.3 card to Web Chat < 4.12.0, they will see differences when they are sending the same card to 4.12.0, despite Web Chat 4.12.0 is rendering a more truthful version of the card.

We should add a "parse the card up to this version" options to allow the developers to (temporarily) downgrade the Adaptive Cards parser, giving developers more time to fix/update their cards.

[feature-request]

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #3777.

### Added

-  Resolves [#3777](https://github.com/microsoft/BotFramework-WebChat/issues/3777). Added a new `adaptiveCardsParserMaxVersion` style options for selecting the maximum supported version when parsing an Adaptive Cards, by [@compulim](https://github.com/compulim) in PR [#3778](https://github.com/microsoft/BotFramework-WebChat/pull/3778)

## Description

Adding a new `adaptiveCardsParserMaxVersion` to style options to select the "maximum version of schema to use when parsing Adaptive Cards".

## Design

As mentioned in #3777, this option will give developers more time before they can tweak their cards to render truthfully with the new Adaptive Cards 2.5.0 renderer, which support card schema up to 1.3.

## Specific Changes

- Added a new `adaptiveCardsParserMaxVersion` style options to full and ES5 bundle
- Added tests

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] ~I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [x] ~Accessibility reviewed (tab order, content readability, alt text, color contrast)~
-  [x] ~Browser and platform compatibilities reviewed~
-  [x] ~CSS styles reviewed (minimal rules, no `z-index`)~
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] ~Internationalization reviewed (strings, unit formatting)~
-  [x] ~`package.json` and `package-lock.json` reviewed~
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/bundle/src/adaptiveCards/hooks/internal/useParseAdaptiveCardJSON.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { hooks } from 'botframework-webchat-component';
-import { useCallback } from 'react';
+import { useCallback, useMemo } from 'react';
 
 import useAdaptiveCardsPackage from '../useAdaptiveCardsPackage';
 
-const { useDirection } = hooks;
+const { useDirection, useStyleOptions } = hooks;
```
Comment (corinagum) [inline packages/bundle/src/adaptiveCards/hooks/internal/useParseAdaptiveCardJSON.js:6]: Does this need a `defaultStyleOptions.js` change?

Comment (compulim) [reply packages/bundle/src/adaptiveCards/hooks/internal/useParseAdaptiveCardJSON.js:6]: Discussed offline. We will do a change when we work on type definition file.

**Review by corinagum — APPROVED** [review]
LGTM

**Comment by corinagum** [comment]
Patch draft:

## Note to CDN Users

Users linking the CDN `/latest/webchat*.js`, please note that there is a corner case of breaking changes that may apply to your Adaptive Cards. 

In 4.12.0, Web Chat bumped to version 2.5 of Adaptive Cards, which utilizes the 1.3 Adaptive Cards schema. Up until now, any Adaptive Cards in Web Chat using 1.3 schema properties (with card version <1.3), did not see any validation errors occur on their card (if applicable), due to these properties being ignored. With the 4.12.0 bump, any cards (even those marked as version < 1.3) using 1.3 properties will be validated and thrown by Web Chat if exceptions are detected. For example, the property `isRequired` is new to Input.Text, and if set to `true` while missing the required `errorMessage` and/or `label` would create validation errors in Web Chat, regardless of the card's version.

## How to fix - pre-4.12.1 patch

1. Lock down the app’s Web Chat version to /4.11.0/webchat*.js until you are able to verify your Adaptive Card(s) in Adaptive Cards schema 1.3 and Web Chat 4.12.0.
   - For stability purposes, **we recommend this step for all public bots**. Future Web Chat bumps may be done manually to verify updates before production bots are deployed.
1. Verify that all properties used in Adaptive Cards match the schema version of those properties. [Adaptive Cards Schema Explorer](https://adaptivecards.io/explorer/) 
   - The [Adaptive Cards Designer](https://adaptivecards.io/designer/) is a great way to visually verify Adaptive Cards.
3.	After card verification, you may bump to Web Chat 4.12.0 (Or 4.12.1) using `/4.12.0/webchat*.js at your earliest convenience.

Note: The Adaptive Card builder has the `validateProperties()` method which can be used to validate a card during development. This call will return an array of errors if applicable.

## 4.12.1 patch: New style property `adaptiveCardsParserMaxVersion`

Web Chat 4.12.1 patch includes a new style property allowing developers to choose the max Adaptive Cards schema version. See [PR #3778](https://github.com/microsoft/BotFramework-WebChat/pull/3778) for code changes. 

To specify a different max version, you can adjust the style options, shown below:

```js
  window.WebChat.renderWebChat(
    {
      directLine,
      store,
      styleOptions: {
        adaptiveCardsParserMaxVersion: '1.2'
      }
    },
    document.getElementById('webchat')
  );
```

- Web Chat will apply the maximum schema available according to the Adaptive Cards version (as of this patch, schema 1.3) by default.
- An invalid version will revert to Web Chat's default.

**Pull request merged**